catroid/src/main/java/org/catrobat/catroid/ui/ProjectActivity.java

```diff
@@ -212,7 +211,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 				addSpriteFromUri(uri);
 				break;
 			case SPRITE_CAMERA:
-				uri = Uri.fromFile(new File(DEFAULT_ROOT_DIRECTORY, getString(R.string.default_look_name) + ".jpg"));
+				uri = new ImportFromCameraLauncher(this).getCacheCameraUri();
```

Comment:
this would be way nicer when the camera intent would return the uri as a result data instead of making a new importFromCamerLauncher to reproduce the same uri again.